src/fosslight_binary/_jar_analysis.py

```
@@ -3,309 +3,364 @@
-                        oss_name_found = True
-                    # If oss_name is found, break
-                    if oss_name_found:
+
+def merge_binary_list(jar_items, bin_list):
+    not_found_bin = []
+
+    for key, value in jar_items.items():
+        found = False
+        oss_list = value["oss_list"]
+        sha1 = value.get("sha1", "")
+        for bin in bin_list:
```

Ruff (0.15.12)

[error] Variable `bin` is shadowing a Python builtin

(A001)